src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java

```diff
@@ -0,0 +1,155 @@
+        }
+        catch (CheckstyleException ex) {
+            throw new IllegalStateException(ex);
+        }
+        try {
+            CHECKSTYLE_MODULES = CheckUtil.getCheckstyleModules();
+        }
+        catch (IOException ex) {
+            throw new IllegalStateException(ex);
+        }
+    }
+
+    @Override
+    protected ModuleCreationOption findModuleCreationOption(String moduleName) {
+        ModuleCreationOption moduleCreationOption = ModuleCreationOption.IN_CHECKER;
+
+        for (Class<?> moduleClass : CHECKSTYLE_MODULES) {
+            if (moduleClass.getSimpleName().equals(moduleName)
+                    || moduleClass.getSimpleName().equals(moduleName + "Check")) {
+                if (ModuleReflectionUtil.isCheckstyleTreeWalkerCheck(moduleClass)
+                        || ModuleReflectionUtil.isTreeWalkerFilterModule(moduleClass)) {
+                    moduleCreationOption = ModuleCreationOption.IN_TREEWALKER;
+                }
+                break;
+            }
+        }
+
+        return moduleCreationOption;
+    }
+
+    @Override
+    protected DefaultConfiguration createModuleConfig(Class<?> clazz) {
+        return new DefaultConfiguration(clazz.getName());
+    }
+
+    /**
+     * Returns {@link Configuration} instance for the given module name.
+     * This implementation uses {@link #getModuleConfig(String, String)} method inside.
+     * @param moduleName module name.
+     * @return {@link Configuration} instance for the given module name.
+     */
+    protected static Configuration getModuleConfig(String moduleName) {
+        return getModuleConfig(moduleName, null);
+    }
+
+    /**
+     * Returns {@link Configuration} instance for the given module name.
+     * This implementation uses {@link #getModuleConfig(String)} method inside.
+     * @param moduleName module name.
+     * @param moduleId module id.
+     * @return {@link Configuration} instance for the given module name.
+     */
+    protected static Configuration getModuleConfig(String moduleName, String moduleId) {
+        final Configuration result;
+        final List<Configuration> configs = getModuleConfigs(moduleName);
+        if (configs.size() == 1) {
+            result = configs.get(0);
+        }
+        else if (configs.isEmpty()) {
+            throw new IllegalStateException("no instances of the Module was found: " + moduleName);
+        }
+        else if (moduleId == null) {
+            throw new IllegalStateException("multiple instances of the same Module are detected");
+        }
+        else {
+            result = configs.stream().filter(conf -> {
+                try {
+                    return conf.getAttribute("id").equals(moduleId);
+                }
+                catch (CheckstyleException ex) {
+                    throw new IllegalStateException("problem to get ID attribute from " + conf, ex);
+                }
+            })
+            .findFirst().orElseGet(null);
```

Comment:
Should it be something like
```
.findFirst().orElseThrow(
  new IllegalStateException("problem with module config"));
```
? This case should not be possible, but if it happens, it is better to give a tip to the engineer.

Reply:
Done.